### DISCUSSION TITLE
Keep marker shown until touching another point of the screen

### DISCUSSION THREAD

**Discussion by ampeixoto** [patrykandpatrick/vico#356] [Ideas]
Hi!

Is it possible to show the marker when the column is touch and instead of hiding it when the user stop pressing the column, continue to show it until the user presses it again or presses on another column?

From what I see, that is not possible out of the box (please let me know if I am wrong).

If it is not supported, is it something you would plan to add?

Thanks :)

**Comment by Gowsky** [comment]
Hello @ampeixoto, unfortunately, this is not supported. However, it is a feature that we’d like to implement soon.

**Reply by Praveen-Android** [reply]
@Gowsky Seems like this feature isn't there yet. There is some conversation about a possible workaround in the discussion [here](https://github.com/patrykandpatrick/vico/discussions/859). Would greatly appreciate if you could may be give your inputs towards achieving this behavior. Thanks!

**Reply by Gowsky** [reply]
Hello, @Praveen-Android. We'll take a look at that discussion as soon as we can.

**Comment by mak1nt0sh** [comment]
@ampeixoto I'm doing this for linechart, might work for column too:

> Edit: There's a bug in 1.6.6 for the marker not dissapearing when tapped again.
Fixed by: https://github.com/patrykandpatrick/vico/commit/7eaead90ffe513fc987e5e9bc9105626ef302a15

var persistentMarkers by remember { mutableStateOf(emptyMap<Float, Marker>()) }

```
var persistentMarkers by remember { mutableStateOf(emptyMap<Float, Marker>()) }

markerVisibilityChangeListener = object : MarkerVisibilityChangeListener {
                override fun onMarkerMoved(
                    marker: Marker,
                    markerEntryModels: List<Marker.EntryModel>
                ) {
                    persistentMarkers = mapOf(markerEntryModels.first().entry.x to marker)
                }

                override fun onMarkerShown(
                    marker: Marker,
                    markerEntryModels: List<Marker.EntryModel>
                ) {
                    val markerXValue = markerEntryModels.first().entry.x

                    persistentMarkers =
                        if (persistentMarkers.containsKey(markerXValue)) emptyMap() else
                            mapOf(markerXValue to marker)
                }
            }

lineChart(
...
persistentMarkers = persistentMarkers,
...
```

**Comment by 333bkz** [comment]
`    private var prevMarkerTouchPoint: Point? = null

    override fun handleTouchEvent(point: Point?, byZoom: Boolean) {
        if (point == null) {
            if (byZoom) {
                prevMarkerTouchPoint = null
                super.handleTouchEvent(null, true)
            } else {
                super.handleTouchEvent(prevMarkerTouchPoint, false)
            }
        } else {
            prevMarkerTouchPoint = point
            super.handleTouchEvent(point, byZoom)
        }
    }`

**Comment by Gowsky** [comment]
[Vico v2.3.0-alpha.2](https://github.com/patrykandpatrick/vico/releases/tag/v2.3.0-alpha.2) introduces `CartesianMarkerController`.

**Comment by patrickmichalik** [comment]
[Vico 2.3.0](https://github.com/patrykandpatrick/vico/releases/tag/v2.3.0) is now available.
